packages/react/src/components/DataTable/__tests__/DataTable-test.js

```diff
@@ -208,6 +208,31 @@ describe('DataTable', () => {
           'Field 3:B',
         ]);
       });
+      it('should works correctly with `initialSortColumn` and ` initialSortDirection` props', () => {
```

Comment:
I think it would be ideal to have 3 total tests for this:

1. should be initially sorted by the `initialSortDirection` on the `initialSortHeaderKey` column
2. should be initially sorted by `initialSortDirection`
3. should be initially sorted on `initialSortHeaderKey` column